error_handler.go

```diff
@@ -25,3 +25,12 @@ type ErrorHandler interface {
 	// DO NOT CHANGE: any modification will not be backwards compatible and
 	// must never be done outside of a new major release.
 }
+
+// ErrorHandlerFunc is a convenience adapter to allow the use of a function
+// as ErrorHandler.
+type ErrorHandlerFunc func(error)
+
+// Handle implements ErrorHandler.
```

Comment:
```suggestion
// Handle handles the irremediable error by calling the ErrorHandlerFunc itself.
```

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_always_document_public_symbols)